File: Array/Cpp/09_Find_Minimum_in_Rotated_Sorted_Array.cpp
// Find Minimum in Rotated Sorted Array
// Question Link - https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/
// TC - O(logn)

class Solution {
public:
    int findMin(vector<int>& nums) {
        int mini = INT_MAX;

        int low = 0;
        int high = nums.size()-1;
        
        while(low <= high){
            int mid = (low+high)/2;
            if(nums[low] <= nums[mid]){
                mini = min(mini, nums[low]);
                low = mid + 1;
            }
            else{
                high = mid - 1;
                mini = min(mini, nums[mid]);
            }
        }

        return mini;
    }
};

File: Array/Python/09_Find_Minimum_in_Rotated_Sorted_Array.py
# Find Minimum in Rotated Sorted Array
# Question Link - https://leetcode.com/problems/find-minimum-in-rotated-sorted-array/description/
# TC - O(logn)

import sys

class Solution:
    def findMin(self, nums: List[int]) -> int:
        
        mini = sys.maxsize

        low = 0
        high = len(nums) - 1
        
        while(low <= high):
            mid = (low+high)//2

            if(nums[low] <= nums[mid]):
                mini = min(mini, nums[low])
                low = mid + 1
            else:
                mini = min(mini, nums[mid])
                high = mid - 1

        return mini
